adapter/noop2/noop2.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package noop2
+
+import (
+	"fmt"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/golang/protobuf/proto"
+
+	adapter "istio.io/mixer/pkg/adapter"
+	adapter_cnfg "istio.io/mixer/pkg/adapter/config"
+	sample_report "istio.io/mixer/pkg/templates/sample/report"
+)
+
+type (
+	noop2Adapter struct{}
+	noop2Builder struct{}
+)
+
+func (noop2Builder) Name() string { return "noop2" }
+func (noop2Builder) Description() string {
+	return "An adapter that does nothing, just echos the calls made from mixer"
+}
+func (noop2Builder) Close() error { return nil }
```

Comment:
We _might_ want it - the usecase is that the factory creates a long lived connection to a backend and hands that connection to each instance. In that case, we want them to be able to clean up gracefully.